Helm lint fails with helm4

Expected behavior:

`make lint-helm` passes with both helm3 and helm4.

Current behavior:

Helm linting fails because helm4 strictly checks the schema of Chart.yaml, and Teleport use a custom `.version` field to initialize a yaml Anchor.

```
$ head -n1 examples/chart/*/Chart.yaml
==> examples/chart/event-handler/Chart.yaml <==
.version: &version "19.0.0-prealpha.2"

==> examples/chart/tbot-spiffe-daemon-set/Chart.yaml <==
.version: &version "19.0.0-prealpha.2"

==> examples/chart/tbot/Chart.yaml <==
.version: &version "19.0.0-prealpha.2"

==> examples/chart/teleport-cluster/Chart.yaml <==
.version: &version "19.0.0-prealpha.2"

==> examples/chart/teleport-kube-agent/Chart.yaml <==
.version: &version "19.0.0-prealpha.2"

==> examples/chart/teleport-kube-updater/Chart.yaml <==
.version: &version "19.0.0-prealpha.2"

==> examples/chart/teleport-relay/Chart.yaml <==
.version: &version "19.0.0-prealpha.2"


$ helm version
version.BuildInfo{Version:"v4.1.0", GitCommit:"4553a0a96e5205595079b6757236cc6f969ed1b9", GitTreeState:"clean", GoVersion:"go1.25.6", KubeClientVersion:"v1.35"}

$ make lint-helm
...
Rendering reference for chart examples/chart/tbot
Rendering reference for chart examples/chart/tbot-spiffe-daemon-set
 ✅ All references are up-to-date
"$( GOWORK=off CGO_ENABLED=0 go -C /Users/adamcarheden/src/github.com/gravitational/teleport/master/build.assets/tooling tool -n helm-janitor )" lint
Running lint for chart: teleport-cluster
 ❌ Helm linting chart "teleport-cluster" failed with values "examples/chart/teleport-cluster/.lint/acme-off.yaml"
==> Linting examples/chart/teleport-cluster
[WARNING] Chart.yaml: failed to strictly parse chart metadata file
        error unmarshaling JSON: while decoding JSON: json: unknown field ".version"
...
Error: 1 chart(s) linted, 1 chart(s) failed

2026/07/01 09:21:13 linting chart examples/chart/teleport-cluster
        linting with values "examples/chart/teleport-cluster/.lint/acme-off.yaml"
                command helm exited with status 1
                        exit status 1
make: *** [lint-helm] Error 1
```


Bug details:
- Teleport version: Latest master branch
```

$ git remote -v
origin  https://github.com/gravitational/teleport.git (fetch)
origin  https://github.com/gravitational/teleport.git (push)

$ git log -1 | head -n3
commit 8a67cffc31c6d6b884ab314a425b10c7351a41c3 (HEAD -> master, origin/master, origin/HEAD)
Author: Nick Marais <nicholas.marais@goteleport.com>
Date:   Wed Jul 1 15:33:36 2026 +0100
```

- Recreation steps: run `make lint-helm` with helm4


We could simply set the Anchor at `version` instead of the custom `.version` field. But we'd need to update helm-janitor too:

```
$ grep -r '\.version' build.assets/tooling/cmd/helm-janitor/version.go 
build.assets/tooling/cmd/helm-janitor/version.go:var versionRegex = regexp.MustCompile(`\.version: .*`)
build.assets/tooling/cmd/helm-janitor/version.go:       newChartYaml := versionRegex.ReplaceAll(chartYaml, []byte(fmt.Sprintf(`.version: &version %q`, version)))
```

--- a/build.assets/tooling/cmd/helm-janitor/version.go
+++ b/build.assets/tooling/cmd/helm-janitor/version.go
@@ -40,7 +40,7 @@ func updateVersion(ctx context.Context, version string, charts []Chart) error {
 	return nil
 }
 
-var versionRegex = regexp.MustCompile(`\.version: .*`)
+var versionRegex = regexp.MustCompile(`(?m)^version:\s+&version\s+.*`)
 
 func updateChartVersion(ctx context.Context, chart Chart, version string) error {
 	version = strings.TrimPrefix(version, "v")
@@ -48,7 +48,7 @@ func updateChartVersion(ctx context.Context, chart Chart, version string) error 
 	if err != nil {
 		return trace.Wrap(trace.ConvertSystemError(err), "reading Chart.yaml")
 	}
-	newChartYaml := versionRegex.ReplaceAll(chartYaml, []byte(fmt.Sprintf(`.version: &version %q`, version)))
+	newChartYaml := versionRegex.ReplaceAll(chartYaml, []byte(fmt.Sprintf(`version: &version %q`, version)))
 	if bytes.Equal(chartYaml, newChartYaml) {
 		fmt.Printf(" ⚠️ Warning: Chart.yaml unchanged: %q\n", chart.Path)
 	}

--- a/build.assets/tooling/cmd/helm-janitor/version_test.go
+++ b/build.assets/tooling/cmd/helm-janitor/version_test.go
@@ -28,10 +28,8 @@ import (
 
 const (
 	testChartYaml = `
-.version: &version "1.2.3"
-
 name: test-chart
-version: *version
+version: &version "1.2.3"
 appVersion: *version
 
 dependencies:
@@ -39,10 +37,8 @@ dependencies:
     version: *version
 `
 	updatedTestChartYaml = `
-.version: &version "1.2.4-foobar"
-
 name: test-chart
-version: *version
+version: &version "1.2.4-foobar"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/access/datadog/Chart.yaml
+++ b/examples/chart/access/datadog/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-datadog
 description: A Helm chart for the Teleport Datadog Incident Management Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/access/discord/Chart.yaml
+++ b/examples/chart/access/discord/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-discord
 description: A Helm chart for the Teleport Discord Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/access/email/Chart.yaml
+++ b/examples/chart/access/email/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-email
 description: A Helm chart for the Teleport Email Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/access/jira/Chart.yaml
+++ b/examples/chart/access/jira/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-jira
 description: A Helm chart for the Teleport Jira Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/access/mattermost/Chart.yaml
+++ b/examples/chart/access/mattermost/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-mattermost
 description: A Helm chart for the Teleport Mattermost Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/access/msteams/Chart.yaml
+++ b/examples/chart/access/msteams/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-msteams
 description: A Helm chart for the Teleport MsTeams Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/access/pagerduty/Chart.yaml
+++ b/examples/chart/access/pagerduty/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-pagerduty
 description: A Helm chart for the Teleport Pagerduty Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/access/slack/Chart.yaml
+++ b/examples/chart/access/slack/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-slack
 description: A Helm chart for the Teleport Slack Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/event-handler/Chart.yaml
+++ b/examples/chart/event-handler/Chart.yaml
@@ -1,10 +1,8 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-plugin-event-handler
 description: A Helm chart for Teleport Event Handler Plugin
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 
 dependencies:

--- a/examples/chart/lib/teleport-proxy-lib-test/Chart.yaml
+++ b/examples/chart/lib/teleport-proxy-lib-test/Chart.yaml
@@ -1,9 +1,7 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-proxy-lib-test
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: Test harness for teleport-proxy-lib. Not for installation.
 dependencies:

--- a/examples/chart/lib/teleport-proxy-lib/Chart.yaml
+++ b/examples/chart/lib/teleport-proxy-lib/Chart.yaml
@@ -1,9 +1,7 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-proxy-lib
 type: library
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: |
   Library chart containing all the templates to deploy Teleport proxy.

--- a/examples/chart/lib/teleport-util-lib-test/Chart.yaml
+++ b/examples/chart/lib/teleport-util-lib-test/Chart.yaml
@@ -1,10 +1,7 @@
-.version: &version "19.0.0-prealpha.2"
-
-
 apiVersion: v2
 name: teleport-util-lib-test
 type: application
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: Test harness for teleport-util-lib. Not for installation.
 dependencies:

--- a/examples/chart/lib/teleport-util-lib/Chart.yaml
+++ b/examples/chart/lib/teleport-util-lib/Chart.yaml
@@ -1,9 +1,7 @@
-.version: &version "19.0.0-prealpha.2"
-
 apiVersion: v2
 name: teleport-util-lib
 type: library
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: |
   Library chart containing named templates used across multiple charts.

--- a/examples/chart/tbot-spiffe-daemon-set/Chart.yaml
+++ b/examples/chart/tbot-spiffe-daemon-set/Chart.yaml
@@ -1,8 +1,6 @@
-.version: &version "19.0.0-prealpha.2"
-
 name: tbot-spiffe-daemon-set
 apiVersion: v2
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: This chart deploys the Machine & Workload Identity agent, `tbot`, into your Kubernetes cluster and configures it for issuing SPIFFE SVIDs to other Kubernetes workloads.
 icon: https://goteleport.com/static/teleport-symbol-bimi.svg

--- a/examples/chart/tbot/Chart.yaml
+++ b/examples/chart/tbot/Chart.yaml
@@ -1,8 +1,6 @@
-.version: &version "19.0.0-prealpha.2"
-
 name: tbot
 apiVersion: v2
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: This chart deploys an instance of the Machine ID agent, TBot, into your Kubernetes cluster. This allows workloads running within Kubernetes to authenticate to your Teleport cluster.
 icon: https://goteleport.com/static/teleport-symbol-bimi.svg

--- a/examples/chart/teleport-cluster/Chart.yaml
+++ b/examples/chart/teleport-cluster/Chart.yaml
@@ -1,8 +1,6 @@
-.version: &version "19.0.0-prealpha.2"
-
 name: teleport-cluster
 apiVersion: v2
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: Teleport is an access platform for your infrastructure
 icon: https://goteleport.com/static/teleport-symbol-bimi.svg

--- a/examples/chart/teleport-cluster/charts/teleport-operator/Chart.yaml
+++ b/examples/chart/teleport-cluster/charts/teleport-operator/Chart.yaml
@@ -1,8 +1,6 @@
-.version: &version "19.0.0-prealpha.2"
-
 name: teleport-operator
 apiVersion: v2
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: Teleport Operator provides management of select Teleport resources.
 icon: https://goteleport.com/static/teleport-symbol-bimi.svg

--- a/examples/chart/teleport-kube-agent/Chart.yaml
+++ b/examples/chart/teleport-kube-agent/Chart.yaml
@@ -1,8 +1,6 @@
-.version: &version "19.0.0-prealpha.2"
-
 name: teleport-kube-agent
 apiVersion: v2
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: Teleport provides a secure SSH, Kubernetes, database and application remote access solution that doesn't get in the way.
 icon: https://goteleport.com/static/teleport-symbol-bimi.svg

--- a/examples/chart/teleport-kube-updater/Chart.yaml
+++ b/examples/chart/teleport-kube-updater/Chart.yaml
@@ -1,8 +1,6 @@
-.version: &version "19.0.0-prealpha.2"
-
 name: teleport-kube-updater
 apiVersion: v2
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 type: library
 description: |

--- a/examples/chart/teleport-relay/Chart.yaml
+++ b/examples/chart/teleport-relay/Chart.yaml
@@ -1,8 +1,6 @@
-.version: &version "19.0.0-prealpha.2"
-
 name: teleport-relay
 apiVersion: v2
-version: *version
+version: &version "19.0.0-prealpha.2"
 appVersion: *version
 description: This chart deploys the Teleport Relay service.
 icon: https://goteleport.com/static/teleport-symbol-bimi.svg

